pkg/client/snapshot.go

```diff
@@ -0,0 +1,183 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	snapshotv1 "github.com/acorn-io/runtime/pkg/apis/snapshot.storage.k8s.io/v1"
+	"github.com/acorn-io/runtime/pkg/labels"
+	snapshot2 "github.com/acorn-io/runtime/pkg/snapshot"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+var (
+	annotationAccessModes = "original-pvc-access-modes"
+	annotationResources   = "original-pvc-resource"
+	annotationVolumeName  = "original-pvc-volume-name"
+	annotationVolumeClass = "original-pvc-volume-class"
+)
+
+// SnapshotCreate creates a VolumeSnapshot resource in the cluster using the details of the given *PersistentVolumeClaim.
+// Returns a *VolumeSnapshot or error if one occurred.
+func (c *DefaultClient) SnapshotCreate(ctx context.Context, pvc *corev1.PersistentVolumeClaim) (*snapshotv1.VolumeSnapshot, error) {
+	err := snapshot2.CreateSnapshotClass(ctx, c.Client)
```

Comment:
Should we create the snapshot class when we initialize data? like when controller starts? this seems to be pre-configured by admin and controller, not to be created on demand.

Reply:
I tried to do it when the controller starts but it fails due to lack of appropriate permissions within the cluster. Not sure how to fix that...

Reply:
ok, I can help with troubleshooting but we should move this logic to where we created computeclass and other statis resources.

Reply:
No longer necessary given that we're not generating snapshot classes